### ISSUE TITLE
DecompilerHarness: runtime-async no-await method reports false CS0161

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#5377]
## Summary

The identical-source runtime-async fixture added for #4832 exposes a validity
harness false positive: the product renders the `NoAwait` method as valid
`async Task` C#, but `ValidityCheck.Shell` omits `async` and reports `CS0161`.

Owner: DecompilerHarness validity-shell composition.

## Source

```csharp
#pragma warning disable CS1998
public static async Task NoAwait()
{
}
```

The exact same `AsyncFixtures.cs` compiles successfully in both classic and
runtime-async modes. The runtime artifact carries the runtime-async MethodImpl
flag and lowers `NoAwait` to a physical `ret` without a Task value.

## Reproduction

On `main` at `2951499cf`:

```bash
dotnet run --project src/dotnet-inspect -c Release --no-build -- \
  member ILInspector.Decompiler.Fixtures.ClassicAsync.AsyncFixtures NoAwait \
  --library artifacts/bin/ILInspector.Decompiler.Fixtures.RuntimeAsync/release/ILInspector.Decompiler.Fixtures.RuntimeAsync.dll \
  -S "Decompiled Source" --bare
```

Product output:

```csharp
public static async System.Threading.Tasks.Task NoAwait()
{

}
```

This is valid C#. The empty body is legal because runtime-async metadata
requires the `async` method context.

The library report instead says:

```text
validity: CS0161: AsyncFixtures::NoAwait
```

`ValidityCheck.Shell` consults only the pass-installed
`RequiresAsyncBodyModifier` fact. That fact remains false when no await
survives. Fidelity and ReturnToSender already combine it with the imported
runtime-async metadata fact, so their method shells retain `async`.

## Required outcome

Use one typed method-context rule that combines reconstructed classic-async
context with positively known runtime-async metadata. The runtime `NoAwait`
fixture must compile in the validity shell without `CS0161`; an ordinary
Task-returning method without either fact must remain non-async.

## Non-goals

- No product decompilation change.
- No change to state-machine relationship authentication.
- No broad classic-async or runtime-async reconstruction change.

**Comment by richlander** [comment]
Reopened after an accidental merge-keyword closure by fixture-only PR #5408. Its body said the PR did not **fix #5377**, but GitHub recognized the embedded closure keyword despite the negation. #5408 added the measurement fixture only; this product defect remains unresolved.

**Comment by richlander** [comment]
Investigation on current `main` (`2951499cf`) corrected the original diagnosis: the product emits `public static async Task NoAwait() { }`, which is valid. The false `CS0161` is created only by `ValidityCheck.Shell`, which omitted the positively known runtime-async metadata fact. The issue body and title now reflect the harness-only scope.
